esmvaltool/diag_scripts/climate_patterns/plotting.py

```diff
@@ -0,0 +1,195 @@
+# (C) Crown Copyright 2022-2024, Met Office.
+"""Script containing plotting functions for driving scripts.
+
+Author
+------
+Gregory Munday (Met Office, UK)
+"""
+import iris.quickplot as qplt
+import matplotlib.pyplot as plt
+import numpy as np
+
+from esmvalcore.preprocessor import area_statistics
+
+
+def subplot_positions(j):
```

Comment:
I feel these functions could be improved to remove the duplication, but I wouldn't want to hold up the merge of this PR for it.